### ISSUE TITLE
List of packages needing updates is truncated randomly

### ISSUE THREAD

**Issue by CollinChaffin** [chocolatey/ChocolateyGUI#463]
Just went to update all thinking I only had a page of updates, then realized what was going on.  Have now tested on three more Win7x64 machines all with latest version of Choc and GUI.

Here is one of them demonstrating this issue which is pretty serious since, from what I can tell, is rendering the whole "update" functionality of this GUI unusable right now.

````
D:\> choco outdated
Chocolatey v0.10.8
Outdated Packages
 Output is package name | current version | available version | pinned?

autodesk-fusion360|2.0.3133|2.0.3257|true
hwinfo|5.42|5.56|false
hwinfo.install|5.42|5.56|false
icaros|3.0.0|3.0.1|false
imagemagick.app|7.0.6.7|7.0.6.10|false
imagemagick.tool|7.0.4.3|7.0.6.10|false
imgburn|2.5.8.1|2.5.8.20170708|false
io-ninja|3.7.0|3.7.4|false
irfanviewplugins|4.44|4.44.20170319|false
iTunes|12.5.5|12.6.2|false
jsonedit|0.9.19|0.9.24.0|false
lavfilters|0.69|0.70.2|false
lockhunter|3.1.1.20161220|3.2|false
logstash|2.3.4|5.2.1|false
MarkdownMonster|1.4.8|1.5.12|false
mediainfo|0.7.91|0.7.98|false
mediainfo-cli|0.7.91|0.7.98|false
miktex|2.9.6236|2.9.6361|false
miktex.install|2.9.6236|2.9.6361|false
mirc|7.46|7.49|false
mkvtoolnix|14.0.0|15.0.0|false
monitorswitcher|0.5.2.0|0.6.0.0|false
mp3tag|2.80|2.84.1|false
mpc-hc|1.7.11|1.7.13|false
mRemoteNG|1.74.6023.15437|1.75.7008.17907|false
msi.template|1.0.0|1.0.1|false
msiafterburner|4.3.0.9267|4.3.0.926701|false
MySql.Utilities|1.6.4|1.6.5|false
nasm|2.12.02|2.13.01|false
nmap|7.40|7.60|false
nodejs.install|7.7.2|8.4.0|false
notepadplusplus.install|7.4.2|7.5.1|false
npcap|0.78|0.78|false
nssm|2.24.0.20161223|2.24.0.20170619|false
NuGet.CommandLine|3.4.3|4.3.0|false
obs-studio|0.16.5|19.0.3|false
packer|1.0.3|1.0.4|false
pandoc|1.19.2|1.19.2.1|false
path-copy-copy|12.0|14.0|false
pickles|2.14.0|2.16.2|false
picklesui|2.14.0|2.16.2|false
putty|0.67|0.70|false
putty.install|0.67|0.70|false
putty.portable|0.67|0.70|false
python3|3.6.0|3.6.2|false
rawcopy|1.3.3.20160825|1.3.3.20170718|false
rclone|1.34|1.36|false
rdm|12.5.8.0|12.6.2.0|false
reshack|4.5.30|4.5.30.20170526|false
rhash|1.3.4|1.3.5|true
ripgrep|0.4.0|0.6.0|false
ruby|2.3.1|2.4.1.2|false
rufus|2.15|2.16|false
Silverlight|5.1.50428.0|5.1.50905.0|false
slack|1.1.0|2.6.5|false
speccy|1.30.730|1.31.732|false
spotify|1.0.33.106|1.0.60.492|false
sweet-home-3d|5.3|5.4.0.20170429|false
tagscanner|6.0.19|6.0.22|false
take-command|20.0.25|21.0.32|false
teamspeak|3.0.19.4|3.1.6|false
telegram|1.0.2|1.1.17|false
telegram.install|1.0.2|1.1.17|false
tixati|2.51|2.55|false
tortoisegit|2.4.0.2|2.5.0.0|false
treesizefree|3.4.5|4.0.3|false
ultimate-settings-panel|5.2|5.7.1|false
unetbootin|6.2.5|6.5.5|false
universal-usb-installer|1.9.7.0|1.9.7.800|false
vcredist2015|14.0.24215.20160928|14.0.24215.20170201|false
vcxsrv|1.18.3.0|1.19.2.0|false
visualstudiocode|1.13.0|1.15.1|false
wifiinfoview|2.15|2.27|false
winmerge2011.portable|0.2011.005.42701|0.2011.007.02500|false
wnetwatcher|2.05|2.11|false
wnetwatcher.install|2.05|2.11|false
you-get|0.4.626|0.4.775|false
youtube-dl|2017.06.12|2017.09.02|false

Chocolatey has determined 75 package(s) are outdated.
````

Now, let's open this GUI and let it refresh, even do a manual refresh, wait until progress bar is totally idle (even waited five full minutes) before then checking the "show only packages needing updates" checkbox to filter to outdated packages:

![snag_9-2-2017_16-56-28](https://user-images.githubusercontent.com/9301483/29999069-5c0209d8-9000-11e7-96f5-516f52ef3e05.png)

If you note the position of the GUI scroll bar, that is the bottom of the list, and executing against "all" will also be limited to this (apparently randomly) truncated list.  Note that the GUI list DOES match my above list, but only down to (on this run) putty but on other more outdated machines, the truncated GUI list is literally a third of the full list using the choco outdated command myself.  If I close the GUI and reopen it on this machine, I'll get a different result making it even more dangerous because it does not seem consistent or predictable.  I have no idea what could be causing what appears to be some kind of parsing issue but I'll be happy to help test if you need any additional information.

**Comment by CollinChaffin** [comment]
I did some more testing and what's really weird in the list above, I manually uninstalled and removed the last two packages on the list, Putty and Python.  Guess what happens?  Now, the last two updates continue through RCLONE, so two were now continued it those two places, almost like it is the NUMBER of them......but when I counted quickly I thought I got 38 which does not seem like a likely # to be coded anywhere.  Either way, hopefully that testing behavior helps you guys to perhaps pop a lightbulb on this!

**Comment by CollinChaffin** [comment]
Still testing and noticed another confirmed behavior and now I think I see enough string/list oddities that I have seomthing to go on for me to also begin digging through the source code.

As I've manually updated packages on this particular machine, also of definite note - notice the first picture, where the package "Logstash" for some reason was strangely indented (incorrectly) a single leading whitespace char in all columns.  At first, I thought it was just a bad package.....until I continued testing.  But before hand, note on this first capture that the package "mIRC" is normal like all others (except Logstash):

![snag_9-2-2017_18-18-05 crop](https://user-images.githubusercontent.com/9301483/29999577-e4dd70cc-9013-11e7-9698-fb9910a8c63c.png)



Now, after testing and manually updating Logstash thinking perhaps a bad package was causing this on every machine, now notice that about the same half-way point in the package list, now the mIRC package appears with the same sudden bad leading whitespace char in all columns:

![snag_9-2-2017_19-08-52 crop](https://user-images.githubusercontent.com/9301483/29999581-f25ecc96-9013-11e7-99ca-b4283340237f.png)


....So seeing this makes me really think this is a parsing/split/join array/list/dict issue given that whitespace char oddity changing like that.  I'll start digging as well but hopefully this testing helps you guys narrow it down a bit faster.
